core/device_utils.py

```diff
@@ -46,41 +23,48 @@ def get_block_device(device_node):
-    """ Block the USB device by unmounting and powering it off. """
-    print(f"Blocking USB Device: {device_info['vendor_id']} - {device_info['product_id']}")
+def get_lsblk_device(device):
+    """ Get the block device path (e.g., /dev/sda1) for the USB device. """
+    context = pyudev.Context()
 
-    device_node = device_info['device']
-    block_device = get_block_device(device_node)
+    # Find the parent of the device in the USB subsystem
+    device_parent = None
+    for dev in context.list_devices(subsystem='usb'):
+        if dev.device_node == device:
+            device_parent = dev
+            break
 
-    if not block_device:
-        print(f"Unable to identify block device for {device_node}.")
-        return
+    if not device_parent:
+        print(f"Device parent not found for {device}")
+        return None
 
-    vendor_id = device_info['vendor_id']
-    product_id = device_info['product_id']
+    # Look for a block device (partition) whose parent is the USB device
+    for block_device in context.list_devices(subsystem='block', DEVTYPE='partition'):
+        if device_parent in block_device.ancestors:
+            return block_device.device_node
 
-    # The udev rule to block the USB device
-    blocking_command = f'ATTR{{idVendor}}=="{vendor_id}", ATTR{{idProduct}}=="{product_id}", ATTR{{authorized}}="0"'
+    return None
 
-    # Command to append the blocking rule to the udev file
-    echo_command = ['sudo', '-S', 'tee', '-a', '/etc/udev/rules.d/99-usb-blacklist.rules']
 
-    try:
-        # Pass the sudo password and run the echo command to append to the file
-        subprocess.Popen(echo_command, stdin=subprocess.PIPE, stderr=subprocess.DEVNULL)\
-            .communicate(input=f'{sudo_password}\n{blocking_command}\n'.encode())
+def log_usb_device(device_info, decision):
+    """ Log device information with a decision (allow or block). """
+    if not os.path.exists(LOG_FILE):
+        with open(LOG_FILE, 'w') as file:
+            json.dump([], file)
+
+    with open(LOG_FILE, 'r+') as file:
+        try:
+            logs = json.load(file)
+        except json.JSONDecodeError:
+            logs = []
 
-        # Reload udev rules and trigger them to apply changes
-        update_commands = [
-            ['sudo', '-S', 'udevadm', 'control', '--reload'],
-            ['sudo', '-S', 'udevadm', 'trigger']
-        ]
-        for command in update_commands:
-            subprocess.Popen(command, stdin=subprocess.PIPE, stderr=subprocess.DEVNULL)\
-                .communicate(input=f'{sudo_password}\n'.encode())
+        device_info['decision'] = decision  # Add the decision field
+        logs.append(device_info)
 
-        print(f"USB Device: {vendor_id} - {product_id} blocked successfully")
+        file.seek(0)
+        json.dump(logs, file, indent=4)
 
-    except subprocess.CalledProcessError as e:
-        print(f"Failed to block the USB device: {e}")
+
+def disable_auto_mount():
```

Comment:
i would create a `enable_auto_mount` so when ull stop running (either by bad or happy flow) this will make ur program much more easier to debug and use